Documentation/security/network/encryption-wireguard.rst

```diff
@@ -295,6 +295,45 @@ options:
   - LoadBalancer & NodePort XDP Acceleration
   - Direct Server Return (DSR)
 
+Which traffic encrypted
+=======================
+
+The following table denotes which packets are encrypted with WireGuard depending
+on the mode:
+
++------------------------------------+--------------------+-------------------------+
+| Traffic                            | default mode       | node-to-node encryption |
+|                                    |                    | mode                    |
++------------------------------------+--------------------+-------------------------+
+| Pod to remote Pod                  | Y                  | Y                       |
++------------------------------------+--------------------+-------------------------+
+| Pod to remote Pod via L7 proxy or  | Y                  | Y                       |
+| L7 ingress service                 |                    |                         |
++------------------------------------+--------------------+-------------------------+
+| Pod to remote Pod via Service with | Y                  | Y                       |
+| kube-proxy-replacement             |                    |                         |
++------------------------------------+--------------------+-------------------------+
+| Pod to Egress Gateway node         | Y                  | Y                       |
++------------------------------------+--------------------+-------------------------+
+| Client outside cluster to Pod via  | Y                  | Y                       |
+| ServiceService with tunneling and  |                    |                         |
+| kube-proxy-replacement             |                    |                         |
++------------------------------------+--------------------+-------------------------+
+| Client outside cluster to Pod via  | N                  | Y                       |
+| Service with native routing        |                    |                         |
++------------------------------------+--------------------+-------------------------+
+| Pod to remote Node with tunneling  | ?                  | Y                       |
++------------------------------------+--------------------+-------------------------+
+| Pod to remote Node with native     | N                  | Y                       |
++------------------------------------+--------------------+-------------------------+
+| Node to remote Pod with tunneling  | ?                  | Y                       |
++------------------------------------+--------------------+-------------------------+
+| Node to remote Pod with native     | N                  | Y                       |
+| routing                            |                    |                         |
++------------------------------------+--------------------+-------------------------+
+| Node to Node                       | N                  | Y                       |
```

Comment:
I wonder if we should introduce a short glossary clarifying what we understand as a "Pod", "Node" and "Service". Basically mentioning that all of them need to be Cilium-managed and that traffic nodes or pods not managed by Cilium will not be encrypted. Also mentioning that "Node" includes "hostNetwork pod" 

Again, this is obvious to us developers but might not be to a user who is unfamiliar with Cilium.